crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -72,6 +75,17 @@ impl<S: BuildHasher + Default> MapEntities for HashSet<Entity, S> {
         *self = self.drain().map(|e| entity_mapper.get_mapped(e)).collect();
     }
 }
+
+impl MapEntities for BTreeSet<Entity> {
+    fn map_entities<E: EntityMapper>(&mut self, entity_mapper: &mut E) {
+        let temp = mem::take(self);
+        *self = temp
```

Comment:
I think the borrow checker will let you write this without the temporary, if you want: 

```suggestion
        *self = mem::take(self)
```